### PR TITLE
FOTD Optimizations: Unify IDs with other fields, implement `__getitems__`, add tests.

### PR DESCRIPTION
**Pull request by jacobsela**
# Changes
1. refactored `_load_cached_fields` to `_load_fields`
2. the field `id` (that's always loaded to memory) now goes through this path as well
3. unittests
4. implement __getitems__ rather than __getitem__

## Unifying IDs with other fields

startup time test:
- 32 workers
- 20000 samples

### snippet to test

```python
from time import time_ns
import torch
import fiftyone as fo

def dummy_get_item(sample):
    return "foo"

if __name__ == "__main__":
    import torch.multiprocessing as mp
    mp.set_start_method('forkserver')
    mp.set_forkserver_preload(['torch', 'fiftyone'])

    dataset = fo.load_dataset("quickstart")
    torch_dataset = dataset.to_torch(dummy_get_item)
    dataloader = torch.utils.data.DataLoader(
        torch_dataset,
        batch_size=1,
        num_workers=32,
    )

    time_start = time_ns()
    for _ in dataloader:
        break
    time_end = time_ns()

    print(f"Time taken to load first batch: {(time_end - time_start) / 1_000_000} ms")
```

### before:
```
Time taken to load first batch: 39445.790492 ms
```

### after:
```
Time taken to load first batch: 5597.779764 ms
```

### Snippet to test:

```python
from time import time_ns
import torch
import fiftyone as fo

def id_get_item(sample):
    return sample['id']

if __name__ == "__main__":
    import torch.multiprocessing as mp
    mp.set_start_method('forkserver')
    mp.set_forkserver_preload(['torch', 'fiftyone'])

    BATCH_SIZES = [2 ** i for i in range(8)]

    dataset = fo.load_dataset("stanford-dogs")
    torch_dataset = dataset.to_torch(id_get_item)

    for batch_size in BATCH_SIZES:

        dataloader = torch.utils.data.DataLoader(
            torch_dataset,
            batch_size=batch_size,
            num_workers=32,
        )

        time_start = time_ns()
        for _ in dataloader:
            pass
        time_end = time_ns()

        print(f"==========================")
        print(f"Batch size: {batch_size}")
        print(f"Time to iterate dataset: {(time_end - time_start) / 1_000_000_000} s")
        print(f"Average time per sample: {(time_end - time_start) / len(torch_dataset) / 1_000_000_000} s")
        print(f"==========================")
```

### before:
```
==========================
Batch size: 1
Time to iterate dataset: 23.707311036 s
Average time per sample: 0.0011519587481049563 s
==========================
==========================
Batch size: 2
Time to iterate dataset: 19.771405401 s
Average time per sample: 0.0009607096890670554 s
==========================
==========================
Batch size: 4
Time to iterate dataset: 19.544799116 s
Average time per sample: 0.0009496986936831876 s
==========================
==========================
Batch size: 8
Time to iterate dataset: 19.451719468 s
Average time per sample: 0.0009451758730806609 s
==========================
==========================
Batch size: 16
Time to iterate dataset: 19.619153768 s
Average time per sample: 0.0009533116505344995 s
==========================
==========================
Batch size: 32
Time to iterate dataset: 19.764246164 s
Average time per sample: 0.0009603618155490768 s
==========================
==========================
Batch size: 64
Time to iterate dataset: 20.375881652 s
Average time per sample: 0.0009900817129251701 s
==========================
==========================
Batch size: 128
Time to iterate dataset: 20.963901022 s
Average time per sample: 0.001018654082701652 s
==========================
```

### after:
```
==========================
Batch size: 1
Time to iterate dataset: 24.225526209 s
Average time per sample: 0.001177139271574344 s
==========================
==========================
Batch size: 2
Time to iterate dataset: 12.845724885 s
Average time per sample: 0.0006241848826530612 s
==========================
==========================
Batch size: 4
Time to iterate dataset: 8.813464532 s
Average time per sample: 0.0004282538645286686 s
==========================
==========================
Batch size: 8
Time to iterate dataset: 6.333369983 s
Average time per sample: 0.00030774392531584063 s
==========================
==========================
Batch size: 16
Time to iterate dataset: 5.025789733 s
Average time per sample: 0.0002442074700194364 s
==========================
==========================
Batch size: 32
Time to iterate dataset: 4.908745474 s
Average time per sample: 0.00023852018824101068 s
==========================
==========================
Batch size: 64
Time to iterate dataset: 4.241359042 s
Average time per sample: 0.0002060913042759961 s
==========================
==========================
Batch size: 128
Time to iterate dataset: 4.304301925 s
Average time per sample: 0.00020914975340136053 s
==========================

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Enhanced dataset functionality with a more efficient field loading and caching mechanism.
  - Introduced a new method for batch retrieval of samples, simplifying multi-item access.

- **Tests**
  - Added comprehensive tests to ensure the accuracy and reliability of ID handling, caching, and batch sample retrieval.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Comment by jacobsela** [comment]
@manushreegangwar @danielgural Sorry for double PRing, put this in teams accidentally instead of OSS.

**Comment by danielgural** [comment]
LGTM! Nice improvements!

**Review by danielgural — APPROVED** [review]
All good here from my view!

**Review by manushreegangwar — APPROVED** [review]
